app/uk/gov/hmrc/gform/formtemplate/FormTemplatesControllerRequestHandler.scala

```diff
@@ -157,7 +162,35 @@ object FormTemplatesControllerRequestHandler {
         case (None, Some(_), None, None)       => JsSuccess(())
       }
 
-    sectionValidations andKeep jsonValue.transform(
+    val dmsFormIdValidations: JsResult[Unit] =
+      (jsonValue \ "destinations").toOption match {
+        case Some(id) =>
+          val errorMessages = id
+            .as[JsArray]
+            .value
+            .map { jv =>
+              ((jv \ "id").as[String], (jv \ "type").as[String], (jv \ "dmsFormId").toOption.map(_.as[String]))
```

Comment:
`(jv \ "dmsFormId").toOption.map(_.as[String])` could be simplified to `(jv \ "dmsFormId").asOpt[String]`

Reply:
In my code, I have addressed the possibility of missing **dmsFormId** And I am using **missingDmsFormId** message in this case.

Reply:
You dont need to check for missingDmsFormId, its already taken care when you try (jv \ "dmsFormId").as[String]. Why add new code when existing checks already work?